### ISSUE TITLE
Light Color Advanced Widget - default item is missing error

### ISSUE THREAD

**Issue by d0t101101** [igorgladkov/rotini#71]
Having a hard time getting the new light color widget working (and tried a variety of things) - receiving an error "Invalid configuration  / item: 'default' is missing".

Would someone please share an OH items example to use with this?  My intent is to (primarily) have whole-house lighting scenes controlled by this widget in Rotini :)

Thanks,
.

**Comment by igorgladkov** [comment]
@d0t101101 default item for that widget is Colorpicker item without any "{item}" config. Can you share your sitemap and related item config?

**Comment by d0t101101** [comment]
Hey Igor,

Thanks for that - I was able to get it working with a single color item, but it doesn't seem to work with groups.  For example, my main items config has this for my Hue bulbs:

```
Group:Color All_Lights          "All Lights"     <hue>  (All)

Color Light_GF_Bed_Lamp         "Lamp"      <hue>   (All_Lights, GF_Bed, Lights, LightsDimmer)      {hue="5"}
Color Light_GF_Bath_Ceiling     "Lamp"      <hue>   (All_Lights, GF_Bath, Lights, LightsDimmer)     {hue="1"}
Color Light_GF_Office_Desk      "Lamp"      <hue>   (All_Lights, GF_Office, Lights, LightsDimmer)       {hue="4"}
Color Light_GF_Office_Floor     "Lamp"      <hue>   (All_Lights, GF_Office, Lights, LightsDimmer)       {hue="7"}
Color Light_GF_Dining_Floor     "Lamp"      <hue>   (All_Lights, GF_Great, Lights, LightsDimmer)        {hue="8"}
Color Light_GF_Dining_Lamp      "Lamp"      <hue>   (All_Lights, GF_Dining, Lights, LightsDimmer)       {hue="3"}
Color Light_GF_Great_Lamp       "Lamp"      <hue>   (All_Lights, GF_Great, Lights, LightsDimmer)        {hue="2"}   
Color Light_GF_Kitchen_Strip1   "Strip"     <hue>   (All_Lights, GF_Kitchen, Lights, LightsDimmer)  {hue="6"}
Color Light_GF_Kitchen_Strip2   "Strip"     <hue>   (All_Lights, GF_Kitchen, Lights, LightsDimmer)  {hue="10"}
Color Light_GF_Kitchen_Strip3   "Strip"     <hue>   (All_Lights, GF_Kitchen, Lights, LightsDimmer)  {hue="11"}
Color Light_GF_Living_Floor     "Lamp"      <hue>   (All_Lights, GF_Living, Lights, LightsDimmer)       {hue="12"}
Color Light_GF_Living_Iris      "Lamp"      <hue>   (All_Lights, GF_Living, Lights, LightsDimmer)       {hue="9"}
Color Light_GF_Living_Strip     "Strip"     <hue>   (All_Lights, GF_Living, Lights, LightsDimmer)       {hue="13"}
```

And in my Rotini sitemap, the first item works, the second (against the group of lights) does not.

```
    Frame label="All Lights {widget:light-color-advanced}" {
        Colorpicker item=Light_GF_Office_Desk
        //Colorpicker item=All_Lights
    }
```

Using this widget against 'All_Lights' results in the error "Invalid configuration / item: 'default' is missing". 

All the best,
.

**Comment by igorgladkov** [comment]
@d0t101101 Can you run rest API command for "/rest/items/All_Lights?type=json" in the browser at the same time it fails in the app and post response.
I can't reproduce this issue in my setup.

**Comment by d0t101101** [comment]
@igorgladkov, here you are:

http://OHIP:OHPORT/rest/items/All_Lights?type=json

> {"type":"GroupItem","name":"All_Lights","state":"ON","link":"http://OHIP:OHPORT/rest/items/All_Lights","members":[{"type":"ColorItem","name":"Light_GF_Office_Desk","state":"185.142853,13.725491,100.000000","link":"http://OHIP:OHPORT/rest/items/Light_GF_Office_Desk"},{"type":"ColorItem","name":"Light_GF_Great_Lamp","state":"80,80,100","link":"http://OHIP:OHPORT/rest/items/Light_GF_Great_Lamp"},{"type":"ColorItem","name":"Light_GF_Dining_Floor","state":"80,80,100","link":"http://OHIP:OHPORT/rest/items/Light_GF_Dining_Floor"},{"type":"ColorItem","name":"Light_GF_Bed_Lamp","state":"80,80,100","link":"http://OHIP:OHPORT/rest/items/Light_GF_Bed_Lamp"},{"type":"ColorItem","name":"Light_GF_Office_Floor","state":"0.0,0.0,100","link":"http://OHIP:OHPORT/rest/items/Light_GF_Office_Floor"},{"type":"ColorItem","name":"Light_GF_Kitchen_Strip2","state":"80,80,100","link":"http://OHIP:OHPORT/rest/items/Light_GF_Kitchen_Strip2"},{"type":"ColorItem","name":"Light_GF_Kitchen_Strip1","state":"80,80,100","link":"http://OHIP:OHPORT/rest/items/Light_GF_Kitchen_Strip1"},{"type":"ColorItem","name":"Light_GF_Living_Strip","state":"246.35294,100.0,100","link":"http://OHIP:OHPORT/rest/items/Light_GF_Living_Strip"},{"type":"ColorItem","name":"Light_GF_Kitchen_Strip3","state":"80,80,100","link":"http://OHIP:OHPORT/rest/items/Light_GF_Kitchen_Strip3"},{"type":"ColorItem","name":"Light_GF_Bath_Ceiling","state":"80,80,100","link":"http://OHIP:OHPORT/rest/items/Light_GF_Bath_Ceiling"},{"type":"ColorItem","name":"Light_GF_Living_Floor","state":"80,80,100","link":"http://OHIP:OHPORT/rest/items/Light_GF_Living_Floor"},{"type":"ColorItem","name":"Light_GF_Living_Iris","state":"80,80,100","link":"http://OHIP:OHPORT/rest/items/Light_GF_Living_Iris"},{"type":"ColorItem","name":"Light_GF_Dining_Lamp","state":"80,80,100","link":"http://OHIP:OHPORT/rest/items/Light_GF_Dining_Lamp"}]}

Tried a bunch of stuff with this and couldn't get it working with Rotini.  The same group item works fine for controlling all group members via the colorpicker in HABDroid...

**Comment by igorgladkov** [comment]
@d0t101101 Can you change your Group item declaration to be without ":Color". It actually doesn't work for me with that config at all, since Group needs to be in "Group:itemtype:function" format.
After you changed it, can you post "/rest/items/All_Lights?type=json" response again.

**Comment by d0t101101** [comment]
Thanks again, @igorgladkov!

Here is the latest REST response for All_Lights after removing :Color from the item definition:

> {"type":"GroupItem","name":"All_Lights","state":"Undefined","link":"http://OHIP:OHPORT/rest/items/All_Lights","members":[{"type":"ColorItem","name":"Light_GF_Bed_Lamp","state":"80,80,80","link":"http://OHIP:OHPORT/rest/items/Light_GF_Bed_Lamp"},{"type":"ColorItem","name":"Light_GF_Kitchen_Strip1","state":"80,80,80","link":"http://OHIP:OHPORT/rest/items/Light_GF_Kitchen_Strip1"},{"type":"ColorItem","name":"Light_GF_Kitchen_Strip3","state":"80,80,80","link":"http://OHIP:OHPORT/rest/items/Light_GF_Kitchen_Strip3"},{"type":"ColorItem","name":"Light_GF_Dining_Floor","state":"80,80,80","link":"http://OHIP:OHPORT/rest/items/Light_GF_Dining_Floor"},{"type":"ColorItem","name":"Light_GF_Living_Floor","state":"80,80,100","link":"http://OHIP:OHPORT/rest/items/Light_GF_Living_Floor"},{"type":"ColorItem","name":"Light_GF_Living_Iris","state":"80,80,100","link":"http://OHIP:OHPORT/rest/items/Light_GF_Living_Iris"},{"type":"ColorItem","name":"Light_GF_Office_Floor","state":"80,80,80","link":"http://OHIP:OHPORT/rest/items/Light_GF_Office_Floor"},{"type":"ColorItem","name":"Light_GF_Bath_Ceiling","state":"80,80,80","link":"http://OHIP:OHPORT/rest/items/Light_GF_Bath_Ceiling"},{"type":"ColorItem","name":"Light_GF_Office_Desk","state":"44.727272,43.137257,100.000000","link":"http://OHIP:OHPORT/rest/items/Light_GF_Office_Desk"},{"type":"ColorItem","name":"Light_GF_Kitchen_Strip2","state":"80,80,80","link":"http://OHIP:OHPORT/rest/items/Light_GF_Kitchen_Strip2"},{"type":"ColorItem","name":"Light_GF_Dining_Lamp","state":"80,80,80","link":"http://OHIP:OHPORT/rest/items/Light_GF_Dining_Lamp"},{"type":"ColorItem","name":"Light_GF_Great_Lamp","state":"80,80,80","link":"http://OHIP:OHPORT/rest/items/Light_GF_Great_Lamp"},{"type":"ColorItem","name":"Light_GF_Living_Strip","state":"240.71428,100.0,100","link":"http://OHIP:OHPORT/rest/items/Light_GF_Living_Strip"}]}

FYI - the same Rotini error message persists with this item too.

**Comment by igorgladkov** [comment]
Found the bug, it wasn't related to change in response though, rather something overlooked in widget implementation.

**Comment by igorgladkov** [comment]
Fixed in 0.3.26
